digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/ProjectSectionCalculator.scala

```diff
@@ -160,29 +160,33 @@ object ProjectSectionCalculator {
 
     def assignCalibrationPoints(ready: Seq[ProjectLink], unprocessed: Seq[ProjectLink],
                                 calibrationPoints: Map[Long, UserDefinedCalibrationPoint]): Seq[ProjectLink] = {
+      val link = unprocessed.head
+      //checks calibration points which the link had before starting the project
+      val raCPValue = RoadAddressDAO.getLinkCPValue(link.roadAddressId)
+      val raStartCP = raCPValue == CalibrationCode.AtBeginning.value || raCPValue == CalibrationCode.AtBoth.value
```

Comment:
Here you could compare CalibrationCodes instead of Long values.

Reply:
that works. You could have just used queryById and retrive the whole roadaddress data, but that should be slightly faster since it doesn't have to map so many values. the thing is that if someone later wants to change this code and asumes 0 value means no calibration point as it should, but in orelse it could actually mean also that it could not find previous roadaddress.

Reply:
What value would be the best to return or what should happen if a road address could not be found from the database?